test/benchmark-ping-udp.c

```diff
@@ -0,0 +1,161 @@
+/* Copyright libuv project contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+
+/* Run the benchmark for this many ms */
+#define TIME 5000
+
+typedef struct {
+  int pongs;
+  int state;
+  uv_udp_t udp;
+  struct sockaddr_in server_addr;
+} pinger_t;
+
+typedef struct buf_s {
+  uv_buf_t uv_buf_t;
+  struct buf_s* next;
+} buf_t;
+
+static char PING[] = "PING\n";
+
+static uv_loop_t* loop;
+
+static int completed_pingers;
+static unsigned long completed_pings;
+static int64_t start_time;
+
+
+static void buf_alloc(uv_handle_t* tcp, size_t size, uv_buf_t* buf) {
+  static char slab[64 * 1024];
+  buf->base = slab;
+  buf->len = sizeof(slab);
+}
+
+
+static void buf_free(const uv_buf_t* buf) {
+}
+
+
+static void pinger_close_cb(uv_handle_t* handle) {
+  pinger_t* pinger;
+
+  pinger = (pinger_t*)handle->data;
+#if DEBUG
+  fprintf(stderr, "ping_pongs: %d roundtrips/s\n", pinger->pongs / (TIME / 1000));
```

Comment:
Long line (although I suspect this got copied from somewhere else.)

Reply:
Resolved in 478aa6d8

Reply:
Resolved in 478aa6d8